### PR TITLE
Datacube testing

### PR DESCRIPTION
**Pull request by nannau**
Test the script in climpyrical that loads the design value ensemble into a datacube from a NetCDF4 file. Climpyrical currently requires a strict set of keys in order to safely make assumptions about the input data. 

`datacube.py` uses `xarray.load_dataset` to create a dataset object that contains the necessary data to perform the analysis.

The user must know the name of the design value, as found in the NetCDF4 file, in order to properly use this function.

### CODE REVIEW

File: climpyrical/datacube.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        keys (dict, optional): dictionary of required keys in NetCDF4
+            file
+    Returns:
+        ds (xarray Dataset): data cube of assembled ensemble models
+            into a single variable.
+    """
+
+    ds = xr.open_dataset(data_path)
+    actual_keys = set(ds.variables).union(set(ds.dims))
+    keys.add(design_value_name)
```
[the remark this answers is not in the document]
Comment (nannau) [reply climpyrical/datacube.py]: It's not actually "found" - it is user supplied. I want to make sure that `design_value_name` is in the actual keys. By adding it to `keys` I check it against the actual keys found in the dataset.

File: climpyrical/datacube.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        design_value_name (str): name of design value as found in the
+            NetCDF4 file
+        keys (dict, optional): dictionary of required keys in NetCDF4
+            file
+    Returns:
+        ds (xarray Dataset): data cube of assembled ensemble models
+            into a single variable.
+    """
+
+    ds = xr.open_dataset(data_path)
```
[the remark this answers is not in the document]
Comment (nannau) [reply climpyrical/datacube.py]: My assumption here was to let `xarray` handle those types of misuse errors.

[the remark this answers is not in the document]
Comment (nannau) [reply climpyrical/datacube.py]: I've updated the handling of bad input that doesn't rely on `xarray`. See [here](https://github.com/pacificclimate/climpyrical/pull/14/commits/70a2d72964e836938ab10316637b9eaae3316f0d)

[the remark this answers is not in the document]
Comment (jameshiebert) [reply climpyrical/datacube.py]: Yeah, it's absolutely fair (and advisable!) to delegate that exception *raising* to the library that you're using. However, you *should* document that those are potential outcomes of calling this function. And testing those potential outcomes is good practice.

**Comment by nannau** [comment]
Adds several new checks for the ensemble loading functions of `climpyrical` and tests each new check as well as new cases.
* Checks & tests valid datacube path and filetype
* Tests valid keys check
* Checks & tests data validity of empty xarray datasets or datasets with unexpected NaN, or non-monotonically increasing coordinates  
* Tests the shape of several example NetCDF files
* Tests xarray's `open_dataset` function and compares with `read_data`.
* Tests file not found case
* Improves raise documentation for possible behaviour
* Uses `pkg_resources` to load parameterized test data
* Updates .travis.yml to include files using `pkg_resources`
* Tests provide 100% coverage

**Pull request merged**